client/signals/handlers.py

```diff
@@ -0,0 +1,134 @@
+
+
+@receiver(post_save, sender=User)
+def create_and_update_user_on_nodebb(sender, instance, created, update_fields, **kwargs):
+    """
+    Creates a new user at Nodebb side when a new user is created at edX side. OR
+    Update the previous one if some relevant changes occurs.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        created: Flag will be true if new instance created or false if upgraded.
+        update_fields:  Contain frozen_set of the name of fields updated.
+        **kwargs:  All remaining fields.
+    """
+    if created:
+        user_data = {
+            'username': instance.username,
+            'email': instance.email,
+            'joindate': instance.date_joined.strftime('%s')
+        }
+        task_create_user_on_nodebb.delay(**user_data)
+    elif update_fields and 'last_login' not in update_fields:
+        """
+        On login `last_login` field change for ignoring this change we used this check.
+        We are expecting last_login will never changed from django admin panel.
+        """
+        user_data = {
+            'fullname': '{} {}'.format(instance.first_name, instance.last_name)
+        }
+        task_sync_user_profile_info_with_nodebb.delay(username=instance.username, **user_data)
+
+
+@receiver(post_save, sender=UserProfile)
+def sync_user_profile_info_with_nodebb(sender, instance, **kwargs):
+    """
+    If some changed occurs in the User Profile, makes sure that
+    these changes are also made at Nodebb side.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        **kwargs:  All remaining fields.
+    """
+    user = instance.user
+    user_data = {
+        'fullname': instance.name,
+        'location': '{}, {}'.format(
+            instance.city, instance.country.name),
+        'birthday': '01/01/%s' % instance.year_of_birth
+    }
+    task_sync_user_profile_info_with_nodebb.delay(username=user.username, **user_data)
+
+
+@receiver(pre_delete, sender=User)
+def delete_user_from_nodebb(sender, instance, **kwargs):
+    """
+    Deletes the user from NodeBB if it is deleted from edX.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        **kwargs:  All remaining fields.
+    """
+    task_delete_user_from_nodebb.delay(username=instance.username)
+
+
+@receiver(post_save, sender=CourseOverview)
+def create_category_on_nodebb(sender, instance, created, update_fields, **kwargs):
+    """
+    Whenever a new course is created in edX, creates a new category in NodeBB.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        created: Flag will be true if new instance created or false if upgraded.
+        update_fields:  Contain frozen_set of the name of fields updated.
+        **kwargs:  All remaining fields.
+    """
+    if created:
```

Comment:
Why is the `if created:` check necessary?

Reply:
To check whether the instance is newly created or updated. For separating creation or update task.